<launch>

  <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch"/>
<!--
  <node pkg="rviz" type="rviz" name="rviz" args=""/> 
-->

<!--
  <node pkg="sampling_algorithm" type="sampling" name="sampling_node" output="screen">

    <param name="voxel_size_x" value="0.03"/>
    <param name="voxel_size_x" value="0.03"/>
    <param name="voxel_size_x" value="0.07"/>

    <param name="upsamplingStepSize" value="0.1"/>
  </node>
-->

  <node pkg="Lidar_3DOD_2022" type="Lidar_ROI" name="ROI_node" respawn="true">
  <!--패키지 명, type => excutable file, cpp파일에 ROI_node로 되어있음-->
    <!--ROI parameter-->
    <param name="switch_ROI" value="true"/>
    <param name="ROI_xMin" value="0"/>
    <param name="ROI_xMax" value="7"/>
    <param name="ROI_yMin" value="-5.5"/>
    <param name="ROI_yMax" value="5.5"/>
    <param name="ROI_zMin" value="-1.5"/> <!-- ransac auto off (when up to ROI_zMin > -0.2) -->
    <param name="ROI_zMax" value="1.2"/>
  </node>

  <node pkg="Lidar_3DOD_2022" type="Lidar_ransac" name="ransac_node" respawn="true">
    <!--ransac parameter-->
    <param name="switch_UpSampling" value="false"/>
    <param name="switch_RanSaC" value="true"/>
    <param name="ransac_distanceThreshold" value="0.07"/>
  </node>

  <node pkg="Lidar_3DOD_2022" type="Lidar_Clustering" name="Clustering_node" respawn="true" output="screen">
	
    <!--EuclidClustering parameter-->
    <param name="switch_Euclid" value="false"/>
    <param name="EC_eps" value="0.13"/>
    <param name="EC_MinClusterSize" value="10"/>
    <param name="EC_MaxClusterSize" value="100000"/>

    <!--DBSCANclustering parameter-->
    <param name="switch_DBscan" value="true"/>
    <param name="DBscan_eps" value="0.5"/>
    <param name="DBscan_minPts" value="3"/>
    <param name="DB_MinClusterSize" value="5"/>
    <param name="DB_MaxClusterSize" value="10000"/>

    <!--NoiseFiltering parameter-->
    <param name="switch_NoiseFiltering" value="false"/>

    <!--Filter on/off-->    
    <param name="switch_jiwon_filter" value="true"/>
    <param name="switch_DY_filter" value="true"/>

    <!--jiwonFilter parameter-->
    <param name="REMOVE_FACTOR" value="0.2"/>

    <!--DYFilter parameter-->
    <param name="Ransac_Z_ROI" value="-0.75"/>
  </node>

  <node pkg="Lidar_3DOD_2022" type="Lidar_visual" name="visualization_node" output="screen">
    <param name="switch_visual" value="false"/>
  </node>

  <node pkg="Lidar_3DOD_2022" type="Lidar_visual_2D" name="visual_2D_node" output="screen">
    <param name="switch_visual_2D" value="true"/>
    <param name="location" value="$(find Lidar_3DOD_2022)/include/Lidar_3DOD_2022/plane.png"/>
  </node>

  <node pkg="Lidar_3DOD_2022" type="Lidar_delivery_filter" name="delivery_filter_node" output="screen">

  </node>

</launch>
